### ISSUE TITLE
DELL Switch with FW 6.6: Setting port to trunk via RADIUS

### ISSUE THREAD

**Issue by Venice-89** [inverse-inc/packetfence#5537]
With DellOS 6.6 it's possible to set a Port to trunk via RADIUS.
Would be helpfull to use this feature in Packetfence (for example for AccessPoints which needs trunk ports).

**RADIUS Trunk Mode Assignment**
Some network administrators may choose to use a default configuration on all
ports in the network and administer bespoke network policies via RADIUS.
Dell EMC switches support configuration of switchport trunk mode on ports
via RADIUS. In an 802.1X Access-Accept message, the Cisco VSA devicetraffic-class=switch indicates that the connected device is capable of
forwarding traffic from multiple stations using tagged and untagged traffic.
When an Access-Accept message is received that contains the VSA devicetraffic-class=switch, the switch operationally sets the port to trunk mode and
utilizes the RADIUS-assigned VLAN to set the operational native VLAN. If
not present, the port PVID is used to set the operational trunk port native
VLAN. Spanning-tree portfast is operationally disabled on the port. Any
trunk mode configuration on the port is respected.

Page 320 in Dell Configuration Guide
https://downloads.dell.com/manuals/all-products/esuprt_ser_stor_net/esuprt_networking/esuprt_net_fxd_prt_swtchs/networking-n3000e-on_user%27s-guide13_en-us.pdf

**Comment by fdurand** [comment]
It's exactly how the cisco switches works too.
We are looking for https://tools.ietf.org/html/rfc4675 for floating device concept.
Btw you can already use this attribute but using a radius filter that reply this attribute.

**Comment by Venice-89** [comment]
Maybe I am wrong, but I think the easiest way would be not to rework the floating device integration 
(I think thats working via SNMP? )

Would it not be much easier to add a flag "trunk" to the nodes, like the voice flag.
Then it would be easy to recycle the code from that.
It looks like very similar how that work. 

Just a idea :) 

But for now I will try it like you said via radius filter :) thanks !

[aerohive]
filter = bypass_role
operator = is
value = AerohiveTrunk

[1:aerohive]
scope=returnRadiusAccessAccept
merge_answer=no
answer1 = Cisco-AVPair => devicetraffic-class=switch

**Comment by fdurand** [comment]
Yes good idea for the trunk flag, the device role can assign the native vlan and the trunk flag will set it as trunk.
